sound/soc/intel/boards/sof_sdw_rt5682.c

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2020 Intel Corporation
+
+/*
+ *  sof_sdw_rt5682 - Helpers to handle RT5682 from generic machine driver
+ */
+
+#include "sof_sdw_common.h"
```

Comment:
I think in its latest version this header isn't including all the headers "commonly needed" by all, each .c file is now expected to include all the appropriate headers itself. See #1889

Reply:
> I think in its latest version this header isn't including all the headers "commonly needed" by all, each .c file is now expected to include all the appropriate headers itself. See #1889

Sure Shall correct this..  Thanks for Pointing this out.

Reply:
> Apologize for posting RT5682 machine driver patches before PR:1889 Merge.
> 
> Wanted to discuss and get guidance about the following error that I observe with the patch-rebased to PR:1889, the same issue was not observed with PR:1885
> 
> 1. Jack Detection works fine
> 2. Observe following error during playback
>    LOG
>    [   94.622100] intel-sdw sdw-master-0: failed to set stream pointer on codec dai rt5682-aif1
>    [   94.622108] intel-sdw sdw-master-0: ASoC: can't open interface SDW0 Pin2: -524
>    [   94.622116]  SDW0-Playback: ASoC: BE open failed -524
>    [   94.622121]  SDW0-headphone: ASoC: failed to start some BEs -524
>    [   94.622128]  SDW0-headphone: ASoC: BE playback disconnect check for SDW0-Playback
>    [   94.622130]  SDW0-headphone: freed DSP playback path SDW0-headphone -> SDW0-Playback
>    [   94.622258] rt5682 sdw:0:25d:5682:0: Parity error detected
>    [   94.622519] intel-sdw sdw-master-0: Slave status change
>    [  112.360613]  SDW0-headphone: ASoC: found 0 audio playback paths
>    [  112.360619]  SDW0-headphone: ASoC: SDW0-headphone no valid playback route
>    [  112.360623] sof_sdw sof_sdw: ASoC: find BE for widget ALH2.OUT
>    [  112.360625] sof_sdw sof_sdw: ASoC: try BE : ALH2.OUT
>    [  112.360628]  SDW0-headphone: connected new DPCM playback path SDW0-headphone -> SDW0-Playback
>    [  112.360651]  SDW0-headphone: ASoC: found 1 new BE paths
>    [  112.360653]  SDW0-Playback: ASoC: open playback BE SDW0-Playback

Many Thanks @RanderWang for pointing out the below miss.
Code:
```
static struct sof_sdw_codec_info codec_info_list[] = {
        {
                                .id = 0x5682,
                .direction = {true, true},
-               .dai_name = "rt5682-aif1",
+               .dai_name = "rt5682-sdw",

```
Now I did a Loopback and was able to capture the audio. Shall update the PR and wait for PR:1889 merge.